html/browsers/history/the-location-interface/location-protocol-setter-sameish.html

```diff
@@ -0,0 +1,26 @@
+<!doctype html>
+<title>Set location.protocol to the scheme it already was</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<iframe src=/common/blank.html></iframe>
+<iframe src=/common/blank.html></iframe>
+<iframe src=/common/blank.html></iframe>
+<iframe src=/common/blank.html></iframe>
+<script>
+self.onload = () => {
+  [
+    "http",
+    "ht\x0Atp",
+    "http\x0A",
+    "\x09ht\x09\x0AtP"
+  ].forEach((val, index) => {
+    async_test(t => {
+      self[index].frameElement.onload = t.step_func_done(() => {
+        assert_equals(self[index].location.protocol, "http:");
+      });
+      self[index].location.protocol = val;
+    }, `Set location.protocol to ${encodeURI(val)} (percent-encoded)`);
```

Comment:
You're actually setting it to the outputted value percent-*de*coded, right?

Note that you might be able to get away with no encoding in the assert message at all; I kind of remember the WPT harness doing some escaping for us.

Reply:
Yeah. It doesn't seem to do any escaping if I remove `encodeURI()`. It does seem like I could also invoke `escape()` but that appears to do the same thing for these inputs.

I'll update the message to "(percent-encoded here for clarity)".